source/connect/tls.txt

```diff
@@ -0,0 +1,272 @@
+``tlsAllowInvalidHostnames`` is set), invalid certificates (when ``tlsAllowInvalidCertificates`` is set), or
+potentially revoked certificates (when ``tlsDisableOCSPEndpointCheck`` or ``tlsDisableCertificateRevocationCheck``
+are set).
+
+Supported Libraries
+-------------------
+
+By default, libmongoc will attempt to find a supported TLS library and enable TLS support. This is controlled
+by the cmake flag ``ENABLE_SSL``, which is set to ``AUTO`` by default. This flag accepts the following values:
+
+- ``AUTO``: Links to the system's native TLS library, or attempts to find OpenSSL. This is the default value.
+- ``DARWIN``: Links to Secure Transport, the native TLS library on macOS.
+- ``WINDOWS``: Links to Secure Channel, the native TLS library on Windows.
+- ``OPENSSL``: Links to OpenSSL (libssl). An optional install path may be specified with ``OPENSSL_ROOT``.
+- ``LIBRESSL``: Links to LibreSSL's libtls. (LibreSSL's compatible libssl may be linked to by setting ``OPENSSL``).
+- ``OFF``: Disables TLS support.
+
+OpenSSL
+~~~~~~~
+
+The MongoDB C Driver uses OpenSSL on Linux and Unix platforms (besides macOS). Industry best practices
+and some regulations require the use of TLS 1.1 or newer, which requires at least OpenSSL 1.0.1. Use the following command
+to check your OpenSSL version:
+
+.. code-block:: bash
+
+   openssl version
+
+Ensure your system's OpenSSL is a recent version (at least 1.0.1), or use the following command to install a recent version in a
+non-system path and build against it:
+
+.. code-block:: bash
+
+   cmake -DOPENSSL_ROOT_DIR=/absolute/path/to/openssl
+
+When compiled against OpenSSL, the driver will attempt to load the system default certificate store, as configured by the
+distribution. That can be overridden by setting the ``tlsCAFile`` URI option or with the fields ``ca_file`` and ``ca_dir``
+in the `mongoc_ssl_opt_t <{+api-libmongoc+}/mongoc_ssl_opt_t.html>`__.
+
+The Online Certificate Status Protocol (OCSP) is fully supported
+when using OpenSSL 1.0.1+. However, when a ``crl_file`` is set with `mongoc_ssl_opt_t <{+api-libmongoc+}/mongoc_ssl_opt_t.html>`__ and the ``crl_file`` revokes
+the server's certificate, the certificate is considered revoked, even if the certificate has a valid stapled OCSP response.
+  
+.. tip::
+ 
+   For more information about OCSP, see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`__.
+
+LibreSSL / libtls
+~~~~~~~~~~~~~~~~~
+
+The MongoDB C Driver supports LibreSSL through the use of OpenSSL compatibility checks when configured to compile against
+``openssl``. It also supports the new ``libtls`` library when configured to build against ``libressl``.
+
+When compiled against the Windows native libraries, the ``crl_file`` option of a `mongoc_ssl_opt_t <{+api-libmongoc+}/mongoc_ssl_opt_t.html>`__ is not supported,
+and will issue an error if used.
+
+Setting ``tlsDisableOCSPEndpointCheck`` and ``tlsDisableCertificateRevocationCheck`` has no effect.
```

Comment:
Q: setting these options has no effect when compliling against the Windows native libraries, or always? If it's the former, I'd combine this sentence with the paragraph before